xds/src/main/java/io/grpc/xds/internal/sds/XdsServer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.Server;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.xds.XdsClientWrapperForServerSds;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Wraps a {@link Server} delegate and {@link XdsClientWrapperForServerSds} and intercepts
+ * {@link Server#shutdown()} etc to properly shut down the {@link XdsClientWrapperForServerSds}
```

Comment:
when do we need to have  `XdsClientWrapperForServerSds`? it looks like it is only required during ProtocolNegotiation. if this is the case, we can rely on the PN#close.

Reply:
The `XdsClientWrapperForServerSds` is needed for the life of the server. Note that every incoming connection to the server results into TLS handshake and protocol neg and should use the latest cert/key material pointed by the tlsContext. Why close is on PN#close?

Reply:
okay. that's why i am confused. the ownership of XdsClientWrapperForServerSds is weird. XdsServer should own the XdsClientWrapperForServerSds, and PN should borrow it from the XdsServer. because PN only exists during protocol negotiation. 
but, this code create it in PN. and the ownership is transferred to XdsServer. if we change PN to accept XdsClientWrapperForServerSds from server side the code will be much easier to follow. also, PN doesn't need to create/handle exception of the XdsClientWrapperForServerSds.

Reply:
Just realized the PN exists for the whole lifetime of the server: you set it on the server-builder with `NettyServerBuilder.protocolNegotiator()` and the server calls `newHandler` every time it needs a handshaker. So you were right: I can keep the `XdsClientWrapperForServerSds` with the PN and inside PN#close I can call `XdsClientWrapperForServerSds.shutdown` and I probably don't even need `XdsServer` then.